cli/cmd/play.go

```diff
@@ -118,6 +124,31 @@ func internalPlayModule(cmdCtx *cmdcontext.CmdCtx, args []string) error {
 	os.Setenv("TT_CLI_PLAY_FROM", strconv.FormatUint(playFlags.From, 10))
 	os.Setenv("TT_CLI_PLAY_TO", strconv.FormatUint(playFlags.To, 10))
 
+	// float
+	tsParsed := false
+	floatTimestamp, err := strconv.ParseFloat(playFlags.Timestamp, 64)
+	if err == nil {
+		os.Setenv("TT_CLI_PLAY_TIMESTAMP", strconv.FormatFloat(floatTimestamp, 'f', -1, 64))
+		tsParsed = true
+	}
+	// RFC3339/RFC3339Nano = "2006-01-02T15:04:05.999999999Z07:00"
+	rfc3339NanoTs, err := time.Parse(time.RFC3339Nano, string(playFlags.Timestamp))
+	if err == nil && !tsParsed {
+		tsSec := rfc3339NanoTs.Unix()
+		tsNanoSec := rfc3339NanoTs.Nanosecond()
+		ts := fmt.Sprintf("%s.%s", strconv.FormatInt(tsSec, 10),
+			strconv.FormatInt(int64(tsNanoSec), 10))
+		os.Setenv("TT_CLI_PLAY_TIMESTAMP", ts)
+		tsParsed = true
+	}
+	if !tsParsed {
+		os.Setenv("TT_CLI_PLAY_TIMESTAMP", strconv.FormatUint(math.MaxUint64, 10))
+	}
+	if playFlags.Timestamp != "" && !tsParsed {
+		util.InternalError("Problem with timestamp parsing: %s",
+			version.GetVersion, err)
+	}
```

Comment:
Please, extract the copy-paste code into some common method or function.

Reply:
Fixed.